### ISSUE TITLE
2 byte log file triggers null ptr deref, logswan segfaults

### ISSUE THREAD

**Issue by geeknik** [fcambus/logswan#10]
While fuzzing logswan with American Fuzzy Lop, I was able to trigger a null ptr deref and cause a segfault with logswan and a 2 byte log file.

The log file contains nothing more than :: on a single line.

```
==12377== Invalid read of size 1
==12377==    at 0x4C29514: __strrchr_sse42 (vg_replace_strmem.c:194)
==12377==    by 0x406CB0: parseRequest (parse.c:52)
==12377==    by 0x40255C: main (logswan.c:174)
==12377==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==12377== 
==12377== Process terminating with default action of signal 11 (SIGSEGV)
==12377==  Access not within mapped region at address 0x0
==12377==    at 0x4C29514: __strrchr_sse42 (vg_replace_strmem.c:194)
==12377==    by 0x406CB0: parseRequest (parse.c:52)
==12377==    by 0x40255C: main (logswan.c:174)
==12377==  If you believe this happened as a result of a stack
==12377==  overflow in your program's main thread (unlikely but
==12377==  possible), you can try to increase the size of the
==12377==  main thread stack using the --main-stacksize= flag.
==12377==  The main thread stack size used in this run was 8388608.
Segmentation fault

Program received signal SIGSEGV, Segmentation fault.
__strrchr_sse42 () at ../sysdeps/x86_64/multiarch/strrchr.S:134
134 ../sysdeps/x86_64/multiarch/strrchr.S: No such file or directory.
(gdb) bt
#0  __strrchr_sse42 () at ../sysdeps/x86_64/multiarch/strrchr.S:134
#1  0x0000000000406cb1 in parseRequest ()
#2  0x000000000040255d in main () at /home/geeknik/logswan/src/logswan.c:174
(gdb) i r
rax            0x0  0
rbx            0x0  0
rcx            0x0  0
rdx            0x60cec2 6344386
rsi            0x0  0
rdi            0x0  0
rbp            0x60b3d0 0x60b3d0
rsp            0x7fffffffe238   0x7fffffffe238
r8             0x0  0
r9             0xf47c   62588
r10            0x0  0
r11            0x7ffff74a3190   140737342222736
r12            0x61d870 6412400
r13            0x3  3
r14            0x0  0
r15            0x0  0
rip            0x7ffff74a3210   0x7ffff74a3210 <__strrchr_sse42+128>
eflags         0x10246  [ PF ZF IF RF ]
cs             0x33 51
ss             0x2b 43
ds             0x0  0
es             0x0  0
fs             0x0  0
gs             0x0  0
```

This a bit of an issue as Apache2 logs on the VM that I'm testing are formatted like this:

```
::1 - - [11/Oct/2015:16:55:45 -0500] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.2.22 (Debian) (internal dummy connection)"
::1 - - [11/Oct/2015:16:55:45 -0500] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.2.22 (Debian) (internal dummy connection)"
::1 - - [11/Oct/2015:16:55:45 -0500] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.2.22 (Debian) (internal dummy connection)"
::1 - - [11/Oct/2015:16:55:45 -0500] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.2.22 (Debian) (internal dummy connection)"
```

**Comment by fcambus** [comment]
Thanks for reporting this. I'm not too surprised, I haven't really tried to make the parser bullet-proof yet. I definitely have plans to improve resiliency though.

For info, pasting the four log lines you appended to the issue in a file and processing through logswan gave correct results in my case.

**Comment by fcambus** [comment]
This is fixed in version 1.02, thanks again for reporting!
